kickstarter/android-oss
# EP-490: Android QA CTA Clicked (discover_filter) Phase 2

# 📲 What

- Fix discover filters CTA to send `context_type = "all"` when user selects `All Projects`
- General cleanup, including fixing imports and adding documentation. 
  
# 🤔 Why

Segment integration/QA phase 2 feedback

# 🛠 How

The majority of the code changes in this PR are apart of the cleanup. This is the logic change that handles sending all when all projects is selected. 

![Screen Shot 2021-04-26 at 12 40 30 PM](https://user-images.githubusercontent.com/19390326/116125039-076e3b00-a693-11eb-9a33-0b5b28d4fa60.png)

# 📋 QA

- Open the app
- Tap hamburger button
- Tap on All Projects
- A `CTA Clicked` event should appear in segment with the `context_cta = "discover_filter"`, and `context_type = "all"`

![Screen Shot 2021-04-27 at 11 21 59 AM](https://user-images.githubusercontent.com/19390326/116268236-229c8180-a74b-11eb-85a8-f9f39ca53a82.png)

# Story 📖

[EP-490: Android QA CTA Clicked (discover_filter) Phase 2](https://kickstarter.atlassian.net/browse/EP-490)
